src/attribute.cpp

```diff
@@ -188,6 +182,35 @@ std::vector<uint8_t> Bytes::toVector(const std::string& /*secret*/, const std::a
     return attribute;
 }
 
+ChapPassword::ChapPassword(uint8_t type, const uint8_t* data, size_t size)
+    : Attribute(type),
+      m_value(size - 1)
+{
+    if (size != 17)
+        throw std::runtime_error("Invalid CHAP_PASSWORD attribute size. Should be 17, actual size is " + std::to_string(size));
+
+    m_chapId = data[0];
+
+    for (size_t i = 0; i < size - 1; ++i)
+        m_value[i] = data[i + 1];
+}
+
+ChapPassword::ChapPassword(uint8_t type, uint8_t chapId, const std::vector<uint8_t>& chapValue)
+    : Attribute(type),
+      m_chapId(chapId),
+      m_value(chapValue)
+{
+}
+
+std::vector<uint8_t> ChapPassword::toVector(const std::string& /*secret*/, const std::array<uint8_t, 16>& /*auth*/) const
+{
+    std::vector<uint8_t> attribute(m_value);
+    attribute.insert(attribute.begin(), chapId() % 256);
```

Comment:
Why `chapId()` rather than `m_chapId`?

Also, why `% 256`, this field is `uint8_t`, it never exceeds 255.